### PR TITLE
Add get{Signal}Exporter methods to Simple{Signal}Processor, Batch{Signal}Processor

### PR DESCRIPTION
**Pull request by jack-berg**
Originally discussed in this [comment](https://github.com/open-telemetry/opentelemetry-java/pull/5986#discussion_r1412295799), the idea is to make `AutoConfigurationCustomizer#addSpanProcessorCustomizer` more useful by allowing you to extract the configured SpanExporter/LogRecordExporter from the processor, and plug it into your own processor.
```
autoConfiguration.addSpanProcessorCustomizer(
        (spanProcessor, configProperties) -> {
          if (spanProcessor instanceof BatchSpanProcessor) {
            SpanExporter exporter = ((BatchSpanProcessor) spanProcessor).getExporter();
            return new MyCustomProcessor(exporter);
          }
          return spanProcessor;
        });
```

### CODE REVIEW

**Comment by jkwatson** [comment]
I'm not necessarily opposed to this approach, but it does feel like a band-aid on something that should be easier to do with auto-configuration. 

I don't have a concrete proposal, but what if we had some sort of "autoconfiguration context" that stored all of the currently available auto-configured things, so that any customizer would be able to grab what it needs. It might mean a re-think of the whole auto-configuration codebase/SPIs, though. 🤔

**Comment by jack-berg** [comment]
My thought is that this is sort of the like the pattern of adding [toBuilder](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporter.java#L59-L61) methods. The design question is whether or not we want users to be able to introspect on the configuration of components, or have them be black boxes. Personally, I don't see any advantage to keeping them black boxes. Sure there's more surface area, but its trivial to maintain and hard to imagine abuse. 

> It might mean a re-think of the whole auto-configuration codebase/SPIs, though.

I think the design of autoconfigure and the SPIs is pretty good, and don't have interest in boiling the ocean to solve this problem. One related option I've thought about is to ensure we have public APIs for all the places where we interpret ConfigProperties and return SDK components. For example, we could add a `public static BatchSpanProcessor createProcessor(ConfigProperties, SpanExporter)`, and `public static SpanExporter createSpanExporter(SpanExporter)`. This would allow users to cherry pick portions of autoconfigure they want to use. We already to this to some extent with [ResourceConfiguration#createEnvironmentResource](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ResourceConfiguration.java#L61).

But I'm not inclined to do this without more conversation / motivating use cases because it would require adding a bunch of new surface area.

**Comment by jkwatson** [comment]
👍🏽  before we merge, are we sure the user from [that conversation](https://github.com/open-telemetry/opentelemetry-java/pull/5986#discussion_r1412295799) still needs this?

**Comment by jack-berg** [comment]
> 👍🏽 before we merge, are we sure the user from https://github.com/open-telemetry/opentelemetry-java/pull/5986#discussion_r1412295799 still needs this?

I don't think its strictly necessary since in that case they want to wrap the batch processor instead of replacing it (from what I can gather). But I've personally had use cases where I wanted to replace the batch processor altogether: I wanted to ensure that all spans associated with a particular local root (typically all spans with a particular web request) were guaranteed to be exported in the same batch. This would allow the collector to do a sort of tail based sampling using all spans of request, but which does not require the normal resource intensive / complex routing associated with typical tail sampling where all spans of an entire trace are accumulated before making the decision.

In this case, I am still sending batches of spans to a span exporter, but according to different rules than the batch span processor.

**Review by breedx-splk — APPROVED** [review]
Seems like a reasonable approach to me. 👍🏻

**Comment by jack-berg** [comment]
Will merge this if there are no additional comments.

**Pull request merged**
